cohorts/2023/02-experiment-tracking/homework/train_pipeline.py

```diff
@@ -0,0 +1,241 @@
+"""train_pipeline.py """
+import argparse
+import datetime
+import logging
+import os
+from pathlib import Path
+from typing import Dict
+
+
+import yaml
+
+import optuna
+import mlflow  # type: ignore
+from optuna.samplers import TPESampler
+from sklearn.ensemble import RandomForestRegressor  # type: ignore
+from sklearn.feature_extraction import DictVectorizer  # type: ignore
+from sklearn.metrics import mean_squared_error  # type: ignore
+from sklearn.pipeline import Pipeline  # type: ignore
+
+from datasetsplit import DatasetSplit
+from datahandler import DataHandler
+from mlflowhelper import MlFlowContext, MlFlowExperimentRegistry, MlFlowModelManager
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def run_optimization(
```

Comment:
If in the future, we would like to test further samplers or libraries to optimize, it might be interesting to use a similar ideias as the datahandler class, where we have an abstract and we just inherit from that were we could test further options. If the objective is just this test, I think this is fine. I would move this function to a new file like 'optmization.py' and leave the main functions of the pipeline in this one.